### DISCUSSION TITLE
List of IPs that renovate runs on for VPN access

### DISCUSSION THREAD

**Discussion by ahatzz11** [renovatebot/renovate#16194] [Request Help] [answered]
👋 This is related to #16188, but I thought I would make a separate discussion with a smaller scope to hopefully facilitate some extra conversation. 

We are using the [Renovate Github App](https://github.com/apps/renovate) and trying to configure a private nexus repository that sits behind our internal VPN. We are able to add certain IPs to an allowList though, so I'm just curious if we are able to get the list of IPs that a request from renovate can originate from? I think we could self-host to fix this issue, but if we could keep with the hosted option that would be awesome.

The result would be something similar to how github exposes the [IP addresses for dependabot](https://registry.terraform.io/providers/integrations/github/latest/docs/data-sources/ip_ranges#dependabot), and via the meta api call: https://api.github.com/meta.

**Comment by rarkins** [comment, marked as answer]
We don't currently have a fixed set of IP addresses for the hosted app. The list is ephemeral and changing too quickly to be useful

**Reply by ahatzz11** [reply]
Thanks for an answer! Is this something that would be possible to do in the future?

**Reply by rarkins** [reply]
Yes, it's planned, but potentially not until Q4

**Reply by ido-namely** [reply]
Hi, I'm also interested in accessing a private registry in our VPN. I was wondering if there is an issue I can track for this in case you do work on it in Q4?
Thanks!

**Reply by rarkins** [reply]
No public issue, and it's unfortunately at risk of slipping. I can update here as it progresses

**Reply by ido-namely** [reply]
that would be much appreciated, thank you!

**Reply by ahatzz11** [reply]
+1 - Any updates on this that you can share here would be very much appreciated! Our organization is using self-hosted for now, but would love to move to the official GitHub App.

**Reply by ahatzz11** [reply]
@rarkins Is there any update on this? Officially in Q2 of 2023 so just curious on progress - thanks!

**Reply by rarkins** [reply]
It will likely be completed in the next 4 weeks

**Reply by iwata** [reply]
@rarkins Can you tell me the IP address?

**Reply by rarkins** [reply]
34.239.12.110
3.222.82.38

**Reply by iwata** [reply]
Thank you so much!

**Reply by neilharris1** [reply]
Is this documented anywhere? I mean, do you have these IPs listed somewhere more official than this answer?

**Reply by rarkins** [reply]
No

**Reply by dkhaye** [reply]
We added the above IPs to our whitelist, but now about 1 in 4 of our Renovate runs fails with "...your IP address is not permitted to access this resource".  I suspect there was an additional host(s?) added recently.  Can you share the new IPs?

**Reply by rarkins** [reply]
Hi, please try adding `35.169.49.18`

**Reply by iwata** [reply]
@rarkins We have added the IP addresses mentioned here to our whitelist, but we seem to be frequently accessed from different addresses these days. Have new IPs been added?

**Reply by rarkins** [reply]
More IP's have been added, but the list should be stabilized for around 1-2 years now.

The current list is:

```
    "35.169.49.18",
    "52.207.155.173",
    "52.0.106.131",
    "34.197.147.244",
    "54.243.119.210",
    "98.82.39.166",
    "107.20.215.6",
    "44.206.201.73",
    "34.239.12.110",
    "52.45.235.219",
    "52.31.29.228",
    "34.239.12.110"
```

You can also programmatically follow the "developer-platform" section of https://mend-resources.mend.io/ip.json

**Reply by iwata** [reply]
Thanks!
